include/nanobind/nb_class.h

```diff
@@ -508,6 +562,12 @@ class class_ : public object {
         return *this;
     }
 
+    template <typename Func, typename... Extra>
+    NB_INLINE class_ &def(new_<Func> &&init, const Extra &... extra) {
```

Comment:
could the ``init`` argument just be called ``arg``? Let's also make this change for the `init` and `init_implicit` constructor.